src/test/java/com/clickhouse/kafka/connect/sink/helper/ClickHouseTestHelpers.java

```diff
@@ -142,6 +146,58 @@ public static int countRowsWithEmojis(ClickHouseHelperClient chc, String tableNa
         }
     }
 
+    public static boolean validateRows(ClickHouseHelperClient chc, String topic, Collection<SinkRecord> sinkRecords) {
+        boolean match = false;
+        try (ClickHouseClient client = ClickHouseClient.newInstance(ClickHouseProtocol.HTTP);
+             ClickHouseResponse response = client.read(chc.getServer())
+                     .query(String.format("SELECT * FROM `%s`", topic))
+                     .format(ClickHouseFormat.JSONStringsEachRow)
+                     .executeAndWait()) {
+            Gson gson = new Gson();
+            ClickHouseResponseSummary summary = response.getSummary();
+
+            List<String> records = new ArrayList<>();
+            for (SinkRecord record : sinkRecords) {
+                Map<String, String> recordMap = new TreeMap<>();
+                if (record.value() instanceof HashMap) {
+                    for (Map.Entry<String, Object> entry : ((HashMap<String, Object>) record.value()).entrySet()) {
+                        recordMap.put(entry.getKey(), entry.getValue().toString());
+                    }
+                } else if (record.value() instanceof Struct) {
+                    ((Struct) record.value()).schema().fields().forEach(f -> {
+                        recordMap.put(f.name(), ((Struct) record.value()).get(f).toString());
+                    });
+                }
+
+                String gsonString = gson.toJson(recordMap);
+                records.add(gsonString.replace(".0", "").replace(" ","").replace("'","").replace("\\u003d",":"));
+            }
+
+            List<String> results = new ArrayList<>();
+            LOGGER.info(response.records().toString());
+            response.records().forEach(r -> {
+                String gsonString = r.getValue(0).asString().replace("'","").replace(" ","").replace("\\u003d",":");
+                Map<String, String> resultMap = new TreeMap<>((Map<String, String>) gson.fromJson(gsonString, new TypeToken<Map<String, String>>() {}.getType()));
+                results.add(gson.toJson(resultMap));
+            });
+
+            for (String record : records) {
+                if (results.get(0).equals(record)) {
```

Comment:
We are checking only 1 record, it is interesting to know we expect the retrieve same orders as we inserted what will happen when we are insetting using several workers i think it mainly can impact on our integration tests